app/Models/Event.php

```diff
@@ -70,4 +70,15 @@ public function room()
     {
         return $this->belongsTo(Room::class);
     }
+
+    /**
+     * Whether the event may be edited or deleted.
+     * @return bool
+     */
+    public function isModifiable()
+    {
+        // TODO: Does the room need to be ongoing? Seems harmless to edit future events,
```

Comment:
I don't think it matters as long as the event hasn't passed yet.

Reply:
Following up on this,

This line actually is preventing any events from being modified at all. the `occurs_at` parameter is actually the GameTick number of when the event occurs in the game, and does not correspond to the unix time. So checking `$this->occurs_at->isFuture()` will block all events from being modified.

We can either:
- Add a function in the CLI for the backend to call to verify if the event has happened
- I can just send the unix time for the event instead of the game tick.

Reply:
I would personally be much in favour of the second suggestion, considering the property of "is this event scheduled in the future" is just data and has nothing to do with the inner workings of the engine itself. Storing a tick number implies that the core must use ticks rather than any other method, reducing flexibility.

Either way, right now `occus_at` is a timestamp.

I also was under the impression that perhaps we would get rid of ticks? Or is that still an open debate?

Reply:
Currently, everything in the core engine uses ticks. I think we're going to stick with ticks because if you allow it down to the millisecond you can have pirate+sabo+general combos that recapture the sabo & can deal infinite damage to incoming subs. But I have no problem sending the unix time for game events. I will update the engine to do that once this PR is implemented.

Reply:
I still think we should get rid of ticks, it's more work for us and I think the game would be much more natural and smooth without ticks. Also I feel like ticks are just a way to introduce more potential bugs into the code.

To combat the infinite damage problem, we can just block a nav from rerouting more than once in any 10 minute window, or more than one sub to be sent between the same two outposts in any 10 minute window.